gen/__init__.py

```diff
@@ -464,15 +543,42 @@ def generate(
     if status['status'] == 'errors':
         raise ValidationError(errors=status['errors'], unset=status['unset'])
 
-    argument_dict = {k: v.value for k, v in resolver.arguments.items()}
+    # Gather out the late variables. The presence of late variables changes
+    # whether or not a late package is created
+    late_variables = dict()
+    # TODO(branden): Get the late vars and expressions from resolver.late
+    for source in sources:
+        for setter_list in source.setters.values():
+            for setter in setter_list:
+                if not setter.is_late:
+                    continue
+
+                if setter.name not in resolver.late:
+                    continue
+
+                # Skip late vars that aren't referenced by config.
+                if not resolver.arguments[setter.name].is_finalized:
+                    continue
+
+                # Validate a late variable should only have one source.
```

Comment:
It wouldn't, but this is one of the checks that enforces that. Without this check we could silently allow multiple late expressions for a variable in sources, by selecting whichever gets iterated through last.